### PR TITLE
(MODULES-7150) Refactor Adapter class state management

### PR DESCRIPTION
**Pull request by Iristyle**
Builds on #27, which should be merged prior

- Old code would insert a dummy trigger into brand new tasks
   created for the first time.

   Since they are always immediately removed, and since there is no
   need to insert a dummy trigger, remove that from the code.

### CODE REVIEW

File: lib/puppet_x/puppetlabs/scheduled_task/taskscheduler2_task.rb [hunk 1]
```diff
@@ -51,7 +51,7 @@ def activate(task_name)
     @definition = TaskScheduler2.task_definition(@task)
     @task_password = nil
 
-    @task
```
[the remark this answers is not in the document]
Comment (Iristyle) [reply lib/puppet_x/puppetlabs/scheduled_task/taskscheduler2_task.rb]: Only callers here are tests, and they can be modified at will.

File: lib/puppet_x/puppetlabs/scheduled_task/taskscheduler2_v1task.rb [hunk 2]
```diff
@@ -49,7 +49,7 @@ def activate(task_name)
     @definition = TaskScheduler2.task_definition(@task)
     @task_password = nil
 
-    @task
```
[the remark this answers is not in the document]
Comment (Iristyle) [reply lib/puppet_x/puppetlabs/scheduled_task/taskscheduler2_v1task.rb]: Only callers here are tests, and they can be modified at will.

File: lib/puppet/provider/scheduled_task/taskscheduler_api2.rb [hunk 3]
```diff
@@ -175,7 +170,7 @@ def user=(value)
   end
 
   def create
-    clear_task
+    @triggers   = nil
```
Comment (glennsarti) [inline lib/puppet/provider/scheduled_task/taskscheduler_api2.rb]: nit: double space.

File: lib/puppet/provider/scheduled_task/taskscheduler_api2.rb [hunk 4]
```diff
@@ -14,10 +14,9 @@
 
   def self.instances
     PuppetX::PuppetLabs::ScheduledTask::V2Adapter.tasks.collect do |job_file|
```
Comment (glennsarti) [inline lib/puppet/provider/scheduled_task/taskscheduler_api2.rb]: Can probably rename the iterator here as `task` as job_file has no meaning in the V2 land

File: lib/puppet/provider/scheduled_task/win32_taskscheduler.rb [hunk 5]
```diff
@@ -11,10 +11,9 @@
 
   def self.instances
     PuppetX::PuppetLabs::ScheduledTask::V1Adapter.tasks.collect do |job_file|
```
Comment (glennsarti) [inline lib/puppet/provider/scheduled_task/win32_taskscheduler.rb]: Can probably rename the iterator here as task as job_file has no meaning in the V2 land

File: spec/unit/puppet/provider/scheduled_task/win32_taskscheduler_spec.rb [hunk 6]
```diff
@@ -655,11 +655,9 @@ def time_component
 
   describe '.instances' do
     it 'should use the list of .job files to construct the list of scheduled_tasks' do
```
Comment (glennsarti) [inline spec/unit/puppet/provider/scheduled_task/win32_taskscheduler_spec.rb]: Test name should me modified as `.job` is no longer part of the test

File: lib/puppet/provider/scheduled_task/taskscheduler_api2.rb (hunk 4 above)
Comment (Iristyle) [reply lib/puppet/provider/scheduled_task/taskscheduler_api2.rb]: Done

File: spec/unit/puppet/provider/scheduled_task/win32_taskscheduler_spec.rb (hunk 6 above)
Comment (Iristyle) [reply spec/unit/puppet/provider/scheduled_task/win32_taskscheduler_spec.rb]: Done

**Comment by Iristyle** [comment]
Ok @glennsarti / @michaeltlombardi ... I think we're ready to go now.  I think the secret sauce here was 989c01e.

I have a few other improvement ideas, but we can wait until later on those.

**Comment by glennsarti** [comment]
Passed through adhoc.  Merging.

**Pull request merged**